### DISCUSSION TITLE
Overview of Ideas and Architecture

### DISCUSSION THREAD

**Discussion by passamo9** [passamo9/LaTeX_EnglandWales_Templates#16] [General]
### Introduction
1. First, I would like to thank every contributor: @Giebisch @yjyao and @bond7o6. You guys are absolutely amazing. Could have I imagined the effort, enthusiasm, the work, the productivity that you're all have demonstrated when @Giebisch suggested to start a GitHub? Hell. No.

2. I'm grateful for all of your work. You have no idea.

3. LaTeX may have started in the 20th Century, but it's going to be the standard in the 21st century for any work that's done at university and above. No matter if it's STEM, Humanities, or anything in between.

4. You're not just the future, you're bringing the others into the future. Your contribution here is making an ancient relevant tomorrow.  @Giebisch @yjyao and @bond7o6 your usernames need to be inscribed somewhere in limestone somewhere near Temple Church (established in 1185). Your work is setting a standard. I am so proud of your contributions. Thank you. You're making the survival of the Bar into the 21st century more certain. This means a lot to me, as a barrister called two years ago.

### Call To Action
5. @bond7o6 gave me the advice that a conversation is going to be needed for me and other barristers to give you better advice. I agree. We at the Bar need to start giving you, Contributors, all that you need. You guys have been moving at the speed of light. I mentioned in one thread that it feels like being on velodrome with a tricycle while you, Tour de France athletes are running. We may be on the same track, but we're not on the same lap.

6. The plan is this:

@bond7o6 suggested:

>multi-purpose (legal) class with several layout options

@bond7o6 said we need a more general project direction. I am with you, @bond706. I think we need a space, and better yet, a time for you, the contributors to this topic to cross examine me. You guys get to be the lawyers, and we're all going to have fun doing it.

7. I may call for backup. I don't have a mandate from the Bar of England and Wales. I've approached one barrister this Sunday afternoon, and will try to get a Queen's Counsel in to our call, so there's buy in on my side for the excellent work you're doing. I really cannot emphasise enough what an honour it is to have you contribute. Never in my wildest dreams could have I expected that I would have had motivated contributors in their precious spare time take more than a one second look at my crazy ideas. But you have given me something that I could not have asked from anyone. Faith in the idea that the Bar of England and Wales, beyond their wigs and their gowns, can look to the 20th Century and the 21st Centuries as a time when their work will not just be relevant to justice, but to those that will be reading the documents that they will make to get Justice.

8. I've written far too long to say—Thank you, and I look forward to speaking with you all soon. Let's organise a time for a discussion that will make your contribution a lasting one for the Bar and England and Wales.

**Comment by bond7o6** [comment]
Involving the architecture of the classes/packages. Deciding on a system to follow is probably a good idea. I was thinking of three main ideas.

1. What we currently have:
```
+-- pleading.sty
+-- opinion.sty
+-- etc.sty
```
If we wanted, we could add a shared utility package to keep stuff that impacts multiple templates. You would choose a template by using `\usepackage{template-name}` in the preamble.

2. Separate classes with shared utilities.
```
+-- pleading.cls
+-- opinion.cls
+-- etc.cls
+-- engwales-tools.sty
+-- etc.sty
```
You'd select what template you were using by using `\documentclass{template-name}` in the preamble. 

3. Single class with separate style files.
```
+-- engwales-legal.cls
+-- engwales-tools.sty
+-- pleading.sty
+-- opinion.sty
+-- etc.sty
```
This could be rearranged to exclude all style files by keeping everything inside the class file, but that might be a bit messy. You could also just move `engwales-tools.sty` inside the class file and it'd be all good. You'd select a template by putting `\documentclass[template-name]{engwales-legal}` in the preamble.

I favour option 3, but interested what everyone else thinks. Also other layout ideas are appreciated, I'm likely missing something obvious.

**Reply by passamo9** [reply]
Does idea 3 also take into account the Skeleton template?

**Reply by bond7o6** [reply]
They could all be expanded to include any number of different templates (to a reasonable degree).

**Reply by passamo9** [reply]
That's great news!

**Comment by Giebisch** [comment]
I favour option 3 as well.
~~Would `+-- engwales-tools.sty` be even necessary in this scenario though ? Can't this be included in the class ?~~

**Reply by yjyao** [reply]
+1 for option 3.

> Can't `engwales-tools.sty` be included in the class?

yes and @bond7o6 mentioned this in last paragraph.

**Reply by Giebisch** [reply]
Oops I missed that haha

**Comment by Andonome** [comment]
Maybe it'd be easier to have a style directory?

`style/gdpr.sty`
`style/pleading.sty`

Then examples in the base dir?

**Reply by bond7o6** [reply]
I think that's a good idea, perhaps `tex/` for (vague) compatibility with CTAN, or `engwales-legal/` for ease of use for users (they could just download this folder into their `texmf`). I'm not too convinced about moving the examples down however.
